### PR TITLE
Make context title and summary changeable

### LINKED ISSUE #30: Add ability to display custom track fields in "Context" tab
It would be useful to have the information being displayed in the Context tab be a bit more customisable. 

The way the custom notification message is currently managed would be a perfect way to manage this tab.

### PR DESCRIPTION
**Pull request by gavinhoward**
Closes #30

### CODE REVIEW

File: src/settings/contextsettingspage.ui [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+         <item row="0" column="1">
+          <widget class="QLabel" name="context_item1_label">
+           <property name="sizePolicy">
+            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
+             <horstretch>0</horstretch>
+             <verstretch>0</verstretch>
+            </sizepolicy>
+           </property>
+           <property name="text">
+            <string>Technical Data</string>
```
Comment (jonaski) [inline src/settings/contextsettingspage.ui]: The text for the checkboxes should be set on the checkboxes itself, not in its own label, that way you can click the text to enable it too.

File: src/settings/settingsdialog.cpp [hunk 2]
```diff
@@ -120,6 +121,7 @@ SettingsDialog::SettingsDialog(Application *app, QWidget *parent)
   QTreeWidgetItem *general = AddCategory(tr("General"));
   AddPage(Page_Behaviour, new BehaviourSettingsPage(this), general);
   AddPage(Page_Collection, new CollectionSettingsPage(this), general);
+  AddPage(Page_Context, new ContextSettingsPage(this), general);
```
Comment (jonaski) [inline src/settings/settingsdialog.cpp]: I think we want to place this in the user interface section instead between appearance and notifications.

File: src/widgets/osd.cpp [hunk 3]
```diff
@@ -291,57 +271,74 @@ void OSD::RepeatModeChanged(PlaylistSequence::RepeatMode mode) {
...
+  while ((pos = variable_replacer.indexIn(message, pos)) != -1) {
+    QStringList captured = variable_replacer.capturedTexts();
+    copy.replace(captured[0], ReplaceVariable(captured[0], song, behaviour));
+    pos += variable_replacer.matchedLength();
+  }
+
+  return copy;
+}
+
+QString OSD::ReplaceVariable(const QString &variable, const Song &song, OSD::Behaviour behaviour) {
```
Comment (jonaski) [inline src/widgets/osd.cpp]: Maybe we could move ReplaceVariable to utilities, and keep the OSD::Behaviour stuff in it's own function in  the osd.

**Comment by gavinhoward** [comment]
I have fixed everything you have asked for. Please let me know if there is more.

**Comment by jonaski** [comment]
Thanks!

**Pull request merged**